packages/frontend-2/lib/workspaces/composables/joinRequests.ts

```diff
@@ -0,0 +1,106 @@
+import { useMutation } from '@vue/apollo-composable'
+import {
+  approveWorkspaceJoinRequestMutation,
+  denyWorkspaceJoinRequestMutation
+} from '~/lib/workspaces/graphql/mutations'
+import type {
+  ApproveWorkspaceJoinRequestInput,
+  DenyWorkspaceJoinRequestInput
+} from '~~/lib/common/generated/gql/graphql'
+import { ToastNotificationType, useGlobalToast } from '~~/lib/common/composables/toast'
+import {
+  convertThrowIntoFetchResult,
+  getFirstErrorMessage,
+  modifyObjectField,
+  getCacheId
+} from '~~/lib/common/helpers/graphql'
+
+export const useWorkspaceJoinRequest = () => {
+  const { mutate: approveMutation } = useMutation(approveWorkspaceJoinRequestMutation)
+  const { mutate: denyMutation } = useMutation(denyWorkspaceJoinRequestMutation)
+  const { triggerNotification } = useGlobalToast()
+
+  const approve = async (input: ApproveWorkspaceJoinRequestInput, id: string) => {
```

Comment:
https://www.notion.so/speckle/General-Web-Code-Review-Guide-1f2bf3dfea634460b3778ac3cf4aa1bb?pvs=4#c79ca97569504d1dbd9d5d3b9978360a

also its not obvious what id is - workspace Id? request Id?

Reply:
Fair point, updated 👍